Feat: v8 launch post (#72)
pixijs/pixijs.com: blog/2025-2-15-pixi-v8-launches.md

- **Correcting Past Oversights:** Every project has its lessons. With v8, we've addressed and rearchitected certain aspects of PixiJS, reducing complexity and enhancing functionality, particularly in areas we felt were overengineered in the past (looking at you, textures!).
- **Boosting Performance:** PixiJS is already renowned for its speed. With v8, we've unlocked even greater performance, making it faster across the board compared to v7.

We're incredibly proud of PixiJS v8 and eager to share the improvements and new features with you. While there are some breaking API changes, we've provided a migration guide and ensured compatibility with v7 wherever possible. Get ready to experience the next level of 2D rendering with PixiJS v8!

---

## 🔗 Quick links
- The new Docs for v8 can be found here [LINK TO DOCS]
- [Migration](https://pixijs.com/guides/migrations/v8)
- [Examples](https://pixijs.com/examples)
- [Open Games](https://github.com/pixijs/open-games)

---

## 🎁 Whats New?

So many new things - more than we can easily cover in one post! Here's some head lines, but check out the links above for a more in depth details on the changes!
#### 📦 New Package Structure

> No more "lerna." PixiJS is now just one package with one import root: `import {stuff} from ‘pixi.js’`. This change means we now have much better tree shaking during app compilation, reducing bundle size if not imported.

Reviewer: We kinda abandoned Lerna awhile ago. I think the thing to celebrate here is the complexity of moving away from publishing multiple packages and all the npm headaches, linking locally, and ecosystem issues that caused.
```suggestion
No more multiple packages (e.g., `@pixi/stuff`). PixiJS is now just one package with one import root: `import {stuff} from ‘pixi.js’`. This change means we now have much better tree shaking during app compilation, reducing bundle size if not imported.
```

Reviewer: Also, we do have submodule exports, so you you can do `pixi.js/unsafe-eval` for example for optional things. Not sure if the submodules are worth calling out?

Secondly, the initialization of WebGPU itself is an asynchronous process, so we need to have a promise in there somewhere :)

```ts
import { Application, autoDetectRenderer } from "pixi.js";

const app = new Application();

(async () => {
  await app.init({
    // application options
  });

  // or
  const renderer = await autoDetectRenderer({}); // WebGL or WebGPU

  // do pixi things
})();
```

#### 🖥️ WebGPU Renderer

![PixiJS + webGPU = love](image-1.png)

The obvious one! We have a WebGPU backend to render things. It's worth noting that WebGPU is not inherently faster in all use cases than WebGL (nor is it slower). This is because we are generally more CPU-bound in PixiJS than GPU-bound. However, if you find that your scene has lots of batch breaks (e.g., filters, masks, blend modes), you may find WebGPU will perform faster - especially from the GPU side. WebGPU is still new, and I predict it will get faster with time, just as WebGL did! It's a great starting point!

> #### 📈 New Performance Bar

Reviewer: In journalism this is called "burying the lede". I think You should lead with WebGPU and performance. It's the most impressive and important part of this story. More so than the package structure or Promise-ifying Application/autoDetectRenderer above.

Author: good point! will shift them up!

| 100k sprites (changing scene structure) | ~50ms | ~24ms | <div style={{backgroundColor:'lightgreen', color:'black'}}>108%</div> | ~9ms | ~2ms | <div style={{backgroundColor:'lightgreen', color:'black'}}>350%</div> |

These benchmark numbers are based on the Bunnymark test that you can try yourself!

- [v7 Bunnymark](https://goodboydigital.github.io/pixi-bunnymark/dist/?version=v7&count=100000&renderer=webgpu)
- [v8 Bunnymark - WebGPU](https://goodboydigital.github.io/pixi-bunnymark/dist/?version=v8&count=100000&renderer=webgpu)
- [v8 Bunnymark - WebGL](https://goodboydigital.github.io/pixi-bunnymark/dist/?version=v8&count=100000&renderer=webgl)
- [Repo](https://github.com/GoodBoyDigital/pixi-bunnymark)

#### 🌟 Scene Upgrades

![PixiJS logo](blend-modes.png)

- We have introduced the concept of render groups, allowing containers to have their transform applied by the GPU. This means we can have a true 2D hardware-accelerated camera. Perfect for when you have a large static world that you want to pan and zoom around in! (much like in 3D - you move the Camera around the world, not the world around the camera). This can give really great speed boosts!

> ```

Reviewer: Don't forget language on these triple-tick code blocks
```suggestion
```ts
```

const container = new Container({
  isRenderGroup:true // this containers transform is now handled on the GPU!
}) 
```

- Another cool new change is that now blend modes and tints are inherited, much like transforms and alpha. This means you can now easily tint a container, and all its children will have the tint applied - same for blend modes! its as easy as 
  

> ```

Reviewer: ```suggestion
```ts
```

// will make all the children tinted red!
container.tint = 'red'
// will make all the children have the add blend mode
container.blendMode = 'add'
``` 

Rendering to a texture with antialiasing is now super easy; all you have to do is set the new antialiasing property to true when creating a render texture (or applying a filter) just as you do when creating your renderer! Easy peasy :D

> ```

Reviewer: ```suggestion
```ts
```

  ```

- Gradient fill support has been introduced, currently limited to linear gradients, allowing for more visually engaging designs.

- The new `GraphicsPath` class enables the drawing and sharing of shapes. This feature is particularly useful as it allows for the creation of paths that can then be transformed into Mesh geometry using the `buildGeometryFromPath` function, opening up new possibilities for intricate and detailed graphic designs.

```

const path = 

```

For more information on these graphics upgrades and guidance on how to adapt to the enhanced Graphics API, please refer to the migration guide, or why not jump in and play with some examples!.

60% faster??
< todo - BENCHMARK>

#### 📝 Text Upgrades

Text has been upgraded to allow for better performance and usability! We have also integrated HTMLText into v8 as standard.

BitmapFonts can now be generated on the fly or installed upfront as you prefer. They dynamically add characters as the font's glyphs are required, saving on memory. The layout of bitmap text is almost identical to the layout of the default text now, making it easier to switch between the two depending on your needs.

> ```typescript

Reviewer: ```suggestion
```ts
```
